src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +43,60 @@
      * Андрей - 600
      * София - 900</p>
      */
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
-            throws Exception {
-        return null;
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) throws Exception {
+        int[] salaries = new int[names.length];
+        for (String singleData : data) {
+
+            DataLineInfo dataLineInfo = new DataLineInfo(singleData);
+            if (isNeedDate(dateFrom, dateTo, dataLineInfo.date)) {
+                for (int i = 0; i < names.length; i++) {
+                    if (names[i].equals(dataLineInfo.name)) {
+                        salaries[i] += dataLineInfo.hours * dataLineInfo.payment;
+                    }
+                }
+            }
+        }
+        String report = makeReport(dateFrom, dateTo, names, salaries);
+        return report;
+    }
+
+    private static String makeReport(String dateFrom, String dateTo,
+                                     String[] names, int[] salaries) {
+        StringBuilder result = new StringBuilder();
+
+        result.append("Отчёт за период ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append("\n");
+
+        for (int i = 0; i < names.length; i++) {
```

Comment:
this loop is redundant, you can have this logic implemented in the loops from getSalaryInfo

Reply:
Done. I hope understanded you correctly